lib/3scale/backend/worker_async.rb

```diff
@@ -62,22 +59,24 @@ def process_all
         barrier = Async::Barrier.new
         semaphore = Async::Semaphore.new(@max_concurrent_jobs, parent: barrier)
 
+        Worker.logger.info "Start processing all.."
+
         loop do
           # unblocks when there are new jobs or when .close() is called
           job = @jobs.pop
 
           # If job is nil, it means that the queue is closed. No more jobs are
-          # going to be pushed, so shutdown.
-          shutdown unless job
-
-          break if @shutdown
```

Comment:
This break here was harmful previously, because at this point `job` is present but we do not `perform` it, rather it would just be lost.

Also at this point, it makes no sense to shutdown fetcher, because if queue is closed, then fetcher would be down already.

That leave us just to quit in case queue is closed. `barrier.wait` in the `ensure` block should make sure that all jobs were `performed`.

With this flow we can also remove the `clear_queue` phase. But it should not hurt to stay for now.